### ISSUE TITLE
[Feature Request] Encryption strategy for Distributed token cache

### ISSUE THREAD

**Issue by jmprieur** [AzureAD/microsoft-identity-web#1044]
**Is your feature request related to a problem? Please describe.**
Today the distributed token cache does not encrypt the blob for caches.

**Describe the solution you'd like**
Provide an optional way to enable apps developer to provide/choose an encryption strategy

**Comment by jmprieur** [comment]
One possibility: https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/introduction

Configuration: https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/configuration/overview?view=aspnetcore-5.0

for instance:
```CSharp
   // In startup.cs / or when adding cache
   services.AddDataProtection() 
      .PersistKeysToAzureBlobStorage(dataProtectionConnectionString, dataProtectionContainerName, dataProtectionKeyFile)                
      .ProtectKeysWithAzureKeyVault(new Uri(dataProtectionKeyKeyUri), cred)                
      .SetApplicationName(AuthenticationConstants.ApplicationName);

    // In the controller etc ...
    var dataProtectionProvider = serviceProvider.GetService<IDataProtectionProvider>
```

**Comment by jmprieur** [comment]
Proposed experience.

Customers who want to benefit from cache encryption will add a few lines in their startup.cs:

```CSharp
   services.AddDataProtection() 
      .PersistKeysToAzureBlobStorage(dataProtectionConnectionString, dataProtectionContainerName, dataProtectionKeyFile)                
      .ProtectKeysWithAzureKeyVault(new Uri(dataProtectionKeyKeyUri), cred)                
      .SetApplicationName(AuthenticationConstants.ApplicationName);
```

Proposed design:

- [ ] Inject `IServiceProvider` in the `MsalAbstractTokenCacheProvider` constructor. Have a null default value to avoid a breaking change in existing token caches deriving from `MsalAbstractTokenCacheProvider`  and not provided by us
- [ ] Add a member `dataProtectionProvider` of type `IDataProtectionProvider` computed from the service provider argument when it's not null
- [ ] Before writing, if `dataProtectionProvider` is not null, encrypt the cache

  https://github.com/AzureAD/microsoft-identity-web/blob/cc518815943b3c05be328dd3af88288f4fe8b8c5/src/Microsoft.Identity.Web/TokenCacheProviders/MsalAbstractTokenCacheProvider.cs#L59

- [ ] After reading, if `dataProtectionProvider` is not null, decrypt the cache.
- [ ]  Inject a IServiceProvider in the constructor of all the classes derived from `MsalAbstractTokenCacheProvider`, and that would call the new constructor of  `MsalAbstractTokenCacheProvider`.

**Comment by jennyf19** [comment]
Included in [1.15 release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/1.15.0)
